planet/src/main/webapp/js/itinerary.js

```diff
@@ -224,44 +226,50 @@ function createEventElement(listName, ref, name, address, duration) {
     return eventElement;
 }
 
-function deleteEvent(listName, ref) {
+async function deleteEvent(listName, ref) {
     const userId = firebase.auth().currentUser.uid;
     const eventListRef = database.ref('events/' + userId + '/' + listName);
     const toBeDeletedEventRef = eventListRef.child(ref);
+    let listDate = '08-07-2020'; // Default value
     toBeDeletedEventRef.remove();
 
     // Fix order after deleting the event
     eventListRef.orderByChild('order').once('value', (eventListSnapshot) => {
         let count = 1;
         eventListSnapshot.forEach(function(childEvent) {
-            let event = childEvent.val();
-            let eventKey = childEvent.key;
-            if (event.order !== count) {
-                eventListRef.child(eventKey).update({
-                    order: count
-                });
+            if (childEvent.key !== 'date'){
+                let event = childEvent.val();
+                let eventKey = childEvent.key;
+                if (event.order !== count) {
+                    eventListRef.child(eventKey).update({
+                        order: count
+                    });
+                }
+                count += 1;
+            } else {
+                listDate = childEvent.val();
             }
-            count += 1;
         });
     });
-
-    // Check if this is an event created from a place, and if so, 
-    // remove the user id from the visitors list 
-    const placeRef = database.ref('places/' + ref);
-    placeRef.once('value', (placeSnapshot) => {
-        if (placeSnapshot.val()) {
-            placeRef.child('visitors').child(userId).remove();
-        }
-    });
+    // We check that the list is saved by the user
+    // If the user saved the list than the listDate is not the default value
+    if (listDate !== '08-07-2020') {
```

Comment:
Save '08-07-2020' in a constant.

Reply:
Yeah I though the comment would do, but I'll save it in a constant to make sure is more understandable.

Reply:
Please also address my other comment on the significance of this date.

Reply:
Oh! Sorry didn't see that! 
Actually, supposedly 08-07-2020 is the date we release the application (the presentation day), so we are sure that all the saved lists will have a date different than this because they have to be saved for after this date to be valid.

Reply:
What is the meaning of the |listDate| if it's not the one when the users visit a certain place?

Reply:
It is the date when the users visit a certain place, but it's possible for us not to find this information in the database (for the currentList), that's why I am giving it a default value.

Reply:
Add the TODO in your PR message to the code.

Reply:
It's there ... Line 302 of the itinerary.js file